Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,498 @@
+                if (lobbyPlayer != null)
+                {
+                    lobbyPlayer.transform.SetParent(null);
+                    DontDestroyOnLoad(lobbyPlayer);
+                }
+                else
+                    Debug.LogWarningFormat("OnClientChangeScene: lobbyPlayer is null");
+            }
+            else
+               if (LogFilter.Debug) Debug.LogFormat("OnClientChangeScene {0} {1} {2}", dontDestroyOnLoad, IsClientConnected(), client != null);
+        }
+
+        public override void OnClientSceneChanged(NetworkConnection conn)
+        {
+            if (SceneManager.GetActiveScene().name == LobbyScene)
+            {
+                if (client.isConnected)
+                    CallOnClientEnterLobby();
+            }
+            else
+                CallOnClientExitLobby();
+
+            base.OnClientSceneChanged(conn);
+            OnLobbyClientSceneChanged(conn);
+        }
+
+        #endregion
+
+        #region lobby server virtuals
+
+        public virtual void OnLobbyStartHost() { }
+
+        public virtual void OnLobbyStopHost() { }
+
+        public virtual void OnLobbyStartServer() { }
+
+        public virtual void OnLobbyServerConnect(NetworkConnection conn) { }
+
+        public virtual void OnLobbyServerDisconnect(NetworkConnection conn) { }
+
+        public virtual void OnLobbyServerSceneChanged(string sceneName) { }
+
+        public virtual GameObject OnLobbyServerCreateLobbyPlayer(NetworkConnection conn)
+        {
+            return null;
+        }
+
+        public virtual GameObject OnLobbyServerCreateGamePlayer(NetworkConnection conn)
+        {
+            return null;
+        }
+
+        // for users to apply settings from their lobby player object to their in-game player object
+        public virtual bool OnLobbyServerSceneLoadedForPlayer(GameObject lobbyPlayer, GameObject gamePlayer)
+        {
+            return true;
+        }
+
+        public virtual void OnLobbyServerPlayersReady()
+        {
+            // all players are readyToBegin, start the game
+            ServerChangeScene(GameplayScene);
+        }
+
+        #endregion
+
+        #region lobby client virtuals
+
+        public virtual void OnLobbyClientEnter() { }
+
+        public virtual void OnLobbyClientExit() { }
+
+        public virtual void OnLobbyClientConnect(NetworkConnection conn) { }
+
+        public virtual void OnLobbyClientDisconnect(NetworkConnection conn) { }
+
+        public virtual void OnLobbyStartClient(NetworkClient lobbyClient) { }
+
+        public virtual void OnLobbyStopClient() { }
+
+        public virtual void OnLobbyClientSceneChanged(NetworkConnection conn) { }
+
+        // for users to handle adding a player failed on the server
+        public virtual void OnLobbyClientAddPlayerFailed() { }
+
+        #endregion
+
+        #region optional UI
+
+        public virtual void OnGUI()
```

Comment:
Perhaps make a class that extends NetworkManagerGUI, and adds the lobby specific stuff

Reply:
It's a bare bones default that users are expected to turn off and use their own canvas and such.